### ISSUE TITLE
chmod should not be unconditional in postinst

### ISSUE THREAD

**Issue by cpaelzer** [canonical/ubuntu-pro-client#2722]
In https://github.com/canonical/ubuntu-pro-client/blob/main/debian/ubuntu-advantage-tools.postinst#L447 I've seen that we do chmod this on every install/upgrade.

```
      private_dir="/var/lib/ubuntu-advantage/private"
      if [ -d "$private_dir" ]; then
          chmod 0700 "$private_dir"
      fi
```

While closing LP: #2024204 I thought this should be wrong as well then.
Can we reduce that to only when we create it.
If we need this to be on a particular upgrade make it version bound and only happen on upgrade from the last version it was not done well.

**Comment by cpaelzer** [comment]
I'd appreciate if one could - once this plus lp2024204 are both resolved a test that verifies the permissions "on install" as well as "on re-creation".

**Comment by orndorffgrant** [comment]
Yes this is valid. We should version gate this so it only runs when upgrading from versions <= 19.6 (decided based on looking at the repo state on when this piece of code was committed).
